2009-02-27  Adrien DELLE CAVE <decryptus@proformatique.com>

	pf_sip_mwi_for_thomson_st2030s.patch
	$Revision$
	$Date$

	Correct a misuse of strcasecmp for SIP Notification.

	Replaced phone domain by Asterisk domain in Call-ID of the MWI subscription.
	Special fix for Thomson ST2030S.

Index: asterisk-1.4.24/channels/chan_sip.c
===================================================================
--- asterisk-1.4.24.orig/channels/chan_sip.c	2009-03-17 12:28:03.000000000 +0100
+++ asterisk-1.4.24/channels/chan_sip.c	2009-03-17 12:28:11.000000000 +0100
@@ -16451,6 +16451,20 @@
 	if (peer->mwipvt) {
 		/* Base message on subscription */
 		p = peer->mwipvt;
+
+		if (!strncmp(peer->useragent, "THOMSON ST2030 ", 15) && 
+		    (strstr(peer->useragent, " fw1.65 ")
+		     || strstr(peer->useragent, " fw1.66 ")
+		     || strstr(peer->useragent, " fw2.67 "))) {
+			/* call-id@workaround.proformatique.com */
+			char callid[64];
+			char *stringp;
+
+			ast_copy_string(callid, p->callid, sizeof(callid));
+			stringp = callid;
+			strsep(&stringp, "@");
+			ast_string_field_build(p, callid, "%s@workaround.proformatique.com", callid);
+		}
 	} else {
 		/* Build temporary dialog for this message */
 		if (!(p = sip_alloc(NULL, NULL, 0, SIP_NOTIFY))) 
